prow/lib.sh

```diff
@@ -244,36 +242,40 @@ EOF
         CLUSTERJ_KUBECONFIG="${CLUSTER_KUBECONFIGS[$j]}"
 
         if [ "$j" -gt "$i" ]; then
-          connect_kind_clusters "${CLUSTERI_NAME}" "${CLUSTERI_KUBECONFIG}" "${CLUSTERJ_NAME}" "${CLUSTERJ_KUBECONFIG}"
+          DIRECT_ACCESS=0
+          if [[ "${TOPOLOGY}" == "MULTICLUSTER_SINGLE_NETWORK" ]]; then
+            DIRECT_ACCESS=1
+          fi
+
+          connect_kind_clusters "${CLUSTERI_NAME}" "${CLUSTERI_KUBECONFIG}" "${CLUSTERJ_NAME}" "${CLUSTERJ_KUBECONFIG}" "${DIRECT_ACCESS}"
         fi
       done
     done
-  else
-    # Connect clusters 1 and 2, but leave cluster 3 on a separate network.
-    connect_kind_clusters "${CLUSTER1_NAME}" "${CLUSTER1_KUBECONFIG}" "${CLUSTER2_NAME}" "${CLUSTER2_KUBECONFIG}"
```

Comment:
This configuration would make reusing existing tests a little tricky/make the tests know to use cluster2 or cluster3 for cross-network.  Should we pass in all 3 clusters configs in tests.mk and alter the tests to test connectivity across 3 clusters? That would let us test topologies with a mixture of same-network and cross-network connectivity.

Reply:
yeah, that sgtm